File: app/src/main/java/com/samsung/reliab/feature/main/ui/MainActivity.java
package com.samsung.reliab.feature.main.ui;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.samsung.reliab.R;
import com.samsung.reliab.data.api.sites.SitesApiService;
import com.samsung.reliab.domain.model.Sites;

import retrofit2.Callback;
import retrofit2.Call;
import retrofit2.Response;


public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SitesApiService.getInstance().getName().enqueue(new Callback<Sites>() {
            @Override
            public void onResponse(@NonNull Call<Sites> call, @NonNull Response<Sites> response) {
                assert response.body() != null;
                Log.d("DEBUG", response.body().toString());
            }

            @Override
            public void onFailure(@NonNull Call<Sites> call, @NonNull Throwable throwable) {
                throwable.printStackTrace();
            }
        });

        SitesApiService.getInstance().getUrl().enqueue(new Callback<Sites>() {
            @Override
            public void onResponse(@NonNull Call<Sites> call, @NonNull Response<Sites> response) {
            }

            @Override
            public void onFailure(@NonNull Call<Sites> call, @NonNull Throwable t) {

            }
        });
        SitesApiService.getInstance().getStatus().enqueue(new Callback<Sites>() {
            @Override
            public void onResponse(@NonNull Call<Sites> call, @NonNull Response<Sites> response) {

            }

            @Override
            public void onFailure(@NonNull Call<Sites> call, @NonNull Throwable throwable) {

            }
        });
    }
}

File: app/src/main/java/com/samsung/reliab/feature/main/ui/MainActivity.kt
package com.samsung.reliab.feature.main.ui

import android.os.Bundle
import android.util.Log
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.samsung.reliab.databinding.ActivityMainBinding
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity: AppCompatActivity() {
    private val viewModel: MainViewModel by viewModels()
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        Log.d("MainActivity", "Shit...")

        setContentView(binding.root)

        with(binding) {
            account.textUserInitials.text = "TU"
            account.textUserName.text = "Test User"
            account.textUserEmail.text = viewModel.currentUser?.email ?: "test@test.com"

            imgBtnLogOut.setOnClickListener { viewModel.signOut() }
        }
    }
}